src/coreclr/vm/eepolicy.cpp

```diff
@@ -348,6 +348,14 @@ void LogInfoForFatalError(UINT exitCode, LPCWSTR pszMessage, LPCWSTR errorSource
 {
     WRAPPER_NO_CONTRACT;
 
+    static volatile LONG s_recursionGuard = 0;
+
+    if (::InterlockedCompareExchange(&s_recursionGuard, 1, 0) != 0)
+    {
+        PrintToStdErrA("Repeated call to LogInfoForFatalError skipped to avoid infinite recursion and interaction between crashing threads.");
```

Comment:
This should be user-centric error message, without references to internal implementation details like LogInfoForFatalError

Reply:
Thanks Jan for the suggestion. I have reformulated it to *Repeated fatal crash*, please let me know if you have additional suggestions on improving the message.

Reply:
I would slightly prefer "Fatal error while logging another fatal error" or "Fatal error while logging fatal error". @jkotas what do you prefer?

Reply:
I agree with @janvorli suggestion if we still need error message after addressing the feedback,